static/app/views/performance/transactionSummary/transactionOverview/index.tsx

```diff
@@ -44,68 +48,58 @@ import {
-  transactionThresholdMetric: TransactionThresholdMetric | undefined;
 };
 
-// Used to cast the totals request to numbers
-// as React.ReactText
-type TotalValues = Record<string, number>;
-
-class TransactionSummary extends Component<Props, State> {
-  state: State = {
-    transactionThreshold: undefined,
-    transactionThresholdMetric: undefined,
-    spanOperationBreakdownFilter: decodeFilterFromLocation(this.props.location),
-    eventView: generateSummaryEventView(
-      this.props.location,
-      getTransactionName(this.props.location)
-    ),
-  };
-
-  static getDerivedStateFromProps(nextProps: Readonly<Props>, prevState: State): State {
-    return {
-      ...prevState,
-      spanOperationBreakdownFilter: decodeFilterFromLocation(nextProps.location),
-      eventView: generateSummaryEventView(
-        nextProps.location,
-        getTransactionName(nextProps.location)
-      ),
-    };
+function TransactionOverview(props: Props) {
+  const {api, location, selection, organization, projects} = props;
+  const projectId = decodeScalar(location.query.project);
+  const transactionName = getTransactionName(location);
+
+  if (!defined(projectId) || !defined(transactionName)) {
+    // If there is no transaction name, redirect to the Performance landing page
+    browserHistory.replace({
+      pathname: `/organizations/${organization.slug}/performance/`,
+      query: {
+        ...location.query,
+      },
+    });
+    return null;
   }
 
-  componentDidMount() {
-    const {api, organization, selection} = this.props;
+  const project = projects.find(p => p.id === projectId);
+
+  useEffect(() => {
     loadOrganizationTags(api, organization.slug, selection);
     addRoutePerformanceContext(selection);
-  }
+  }, [selection]);
 
-  componentDidUpdate(prevProps: Props) {
-    const {api, organization, selection} = this.props;
+  const [incompatibleAlertNotice, setIncompatibleAlertNotice] = useState<ReactNode>(null);
+  const handleIncompatibleQuery = (incompatibleAlertNoticeFn, _errors) => {
+    const notice = incompatibleAlertNoticeFn(() => setIncompatibleAlertNotice(null));
+    setIncompatibleAlertNotice(notice);
+  };
 
-    if (
-      !isEqual(prevProps.selection.projects, selection.projects) ||
-      !isEqual(prevProps.selection.datetime, selection.datetime)
-    ) {
-      loadOrganizationTags(api, organization.slug, selection);
-      addRoutePerformanceContext(selection);
-    }
-  }
+  const [transactionThreshold, setTransactionThreshold] = useState<number | undefined>();
+  const [transactionThresholdMetric, setTransactionThresholdMetric] = useState<
+    TransactionThresholdMetric | undefined
+  >();
+
+  const spanOperationBreakdownFilter = decodeFilterFromLocation(location);
 
-  onChangeFilter = (newFilter: SpanOperationBreakdownFilter) => {
-    const {location, organization} = this.props;
+  // TODO: make not undefined
+  const eventView = generateEventView(location, transactionName)!;
```

Comment:
Does this need the `!`? It's def below isn't undefined is it? Did I already ask you this 🤔

Reply:
I added this temporarily and keep forgetting to remove it 😞